core/services/relay/evm/bindings.go

```diff
@@ -3,48 +3,55 @@ package evm
 import (
 	"fmt"
 
-	"github.com/ethereum/go-ethereum/common"
-
-	"github.com/smartcontractkit/chainlink-common/pkg/types"
+	commontypes "github.com/smartcontractkit/chainlink-common/pkg/types"
 )
 
-type Bindings map[string]methodBindings
+type bindings map[string]map[string]binding
 
-func (b Bindings) addEvent(contractName, typeName string, evt common.Hash) error {
-	ae, err := b.getBinding(contractName, typeName, true)
-	if err != nil {
-		return err
+func (b bindings) GetReadBinding(contractName, readName string) (binding, error) {
+	methodReaders, ok := b[contractName]
+	if !ok {
+		return nil, fmt.Errorf("%w: no contract named %s", commontypes.ErrInvalidType, contractName)
 	}
 
-	ae.evt = &evt
-	return nil
-}
-
-func (b Bindings) getBinding(contractName, methodName string, isConfig bool) (*addrEvtBinding, error) {
-	errType := types.ErrInvalidType
-	if isConfig {
-		errType = types.ErrInvalidConfig
-	}
-	methodNames, ok := b[contractName]
+	reader, ok := methodReaders[readName]
 	if !ok {
-		return nil, fmt.Errorf("%w: contract %s not found", errType, contractName)
+		return nil, fmt.Errorf("%w: no readName named %s in contract%s", commontypes.ErrInvalidType, readName, contractName)
 	}
+	return reader, nil
+}
 
-	ae, ok := methodNames[methodName]
+func (b bindings) AddReadBinding(contractName, readName string, reader binding) {
+	mb, ok := b[contractName]
 	if !ok {
-		return nil, fmt.Errorf("%w: method %s not found in contract %s", errType, methodName, contractName)
+		mb = map[string]binding{}
+		b[contractName] = mb
 	}
-
-	return ae, nil
+	mb[readName] = reader
 }
 
-type methodBindings map[string]*addrEvtBinding
-
-func NewAddrEvtFromAddress(address common.Address) *addrEvtBinding {
-	return &addrEvtBinding{addr: address}
+func (b bindings) Bind(boundContracts []commontypes.BoundContract) error {
+	for _, bc := range boundContracts {
+		readings, ok := b[bc.Name]
```

Comment:
in some places these booleans are named more descriptively, lets do it everywhere for consistency

Reply:
which booleans?  ok?  ok is standard convention for reading form a map.

Reply:
I know, but for eg. in `addMethod()` in `chainReader` it is named methodExists instead of ok